src/plugins/util/OverlayPlugin.js

```diff
@@ -11,32 +12,45 @@ module.exports = class OverlayPlugin {
      * @param {Object} instance device renderer instance
      */
     constructor(instance) {
-        this.overlays = [];
-        this.savedState = null;
+        this.widget = null;
+        this.id = generateUID();
```

Comment:
I guess we could use a manual string id in the children classes, that way the store state would be more readable, but the way you've done it is nice too : we don't have to do anything when extending the OverlayPlugin, everything is already done here